src/auth/src/credentials/mds_credential.rs

```diff
@@ -13,37 +13,164 @@
 // limitations under the License.
 
 use crate::credentials::dynamic::CredentialTrait;
-use crate::credentials::{Credential, Result};
+use crate::credentials::{Credential, Result, DEFAULT_UNIVERSE_DOMAIN, QUOTA_PROJECT_KEY};
 use crate::errors::{is_retryable, CredentialError};
 use crate::token::{Token, TokenProvider};
 use async_trait::async_trait;
 use bon::Builder;
 use http::header::{HeaderName, HeaderValue, AUTHORIZATION};
 use reqwest::Client;
+use reqwest::StatusCode;
+use std::default::Default;
 use std::sync::Arc;
 use std::time::Duration;
+use tokio::sync::watch;
+use tokio::sync::Notify;
 
 const METADATA_FLAVOR_VALUE: &str = "Google";
 const METADATA_FLAVOR: &str = "metadata-flavor";
 const METADATA_ROOT: &str = "http://metadata.google.internal/computeMetadata/v1";
 
 pub(crate) fn new() -> Credential {
-    let token_provider = MDSAccessTokenProvider::builder()
-        .endpoint(METADATA_ROOT)
-        .build();
-    Credential {
-        inner: Arc::new(MDSCredential { token_provider }),
-    }
+    MDSCredentialBuilder::default().build()
+}
+
+#[derive(Debug)]
+struct UniverseDomainCache {
+    universe_domain: Option<String>,
+    is_cached: bool,
```

Comment:
I think you should move the caching to a separate PR.  It is hard to reason about two complex changes at the same time. 

With that said: can you add a comment explaining what `is_cached` does?  Why do we need it?